app/payment-links/reference/reference.controller.js

```diff
@@ -50,18 +50,18 @@ function getPage (req, res, next) {
     return next(new NotFoundError('Attempted to access reference page with a product that auto-generates references.'))
   }
 
-  const sessionReferenceNumber = paymentLinkSession.getReference(req, product.externalId)
-
   if (paymentLinkSession.getError(req, product.externalId)) {
     const errors = {}
     const errorMessage = res.locals.__p(paymentLinkSession.getError(req, product.externalId))
     errors[PAYMENT_REFERENCE] = capitaliseFirstLetter(errorMessage)
     paymentLinkSession.setError(req, product.externalId, '')
     data.errors = errors
+    paymentLinkSession.setReference(req, product.externalId, '')
```

Comment:
Rather than setting it to an empty value, might it be better to remove the reference key from the session cookie?

It's probably not how it works currently, but I could see someone potentially checking if the key exists on the cookie and not necessarily checking whether the value is empty or not. It feels slightly better to just entirely remove it?